### ISSUE TITLE
Missing Contrast in medium zoom levels

### ISSUE THREAD

**Issue by dieterdreist** [gravitystorm/openstreetmap-carto#3513]
### Expected behavior
general map that gives indications about the place
### Actual behavior
road focused
### Links and screenshots illustrating the problem
Here some example comparisons between osm-carto and the geofabrik fork, look how the forests are important to understand the structure of the landscape and how they are missing in current osm-carto, until zoom 13 when they become visible but when the scale is so low that you cannot see the bigger picture any more:
![Zoom 8](https://user-images.githubusercontent.com/581288/48547167-47c60e00-e8ca-11e8-8520-115afd49d8d6.png)
![Zoom 9](https://user-images.githubusercontent.com/581288/48547169-47c60e00-e8ca-11e8-89ee-1c4ec2ee8551.png)
![Zoom 10](https://user-images.githubusercontent.com/581288/48547170-485ea480-e8ca-11e8-9b09-3f1c5923078e.png)
![Zoom 11](https://user-images.githubusercontent.com/581288/48547171-485ea480-e8ca-11e8-9532-3fcb1140cf73.png)
![Zoom 12](https://user-images.githubusercontent.com/581288/48547172-485ea480-e8ca-11e8-9ddb-ecfbf763d4b2.png)
![Zoom 13](https://user-images.githubusercontent.com/581288/48547173-48f73b00-e8ca-11e8-9fd6-fdcbf31ecd69.png)

I would say, color fading was an interesting experiment, but in the end it seems the results are worth with the fading than without.

**Comment by kocio-pl** [comment]
My opinion with this experiment is that it works better than other maps. If all the elements are strong, it's making visual mess, especially when the shapes are not simplified and averaged (small elements make noise). In fact we loose some contrast then, because there is less difference between elements (the difference between - say - 10 and 100 is bigger than between 90 and 100).

However fading might be not that fast - it needs some testing to find good balance.

**Comment by kocio-pl** [comment]
I don't follow this fork closely, but I like the idea of generalized green for high/medium/low vegetation very much and after merging #3327 we have main green colors the same as in @imagico fork, so we can go further in this direction. 

Unfortunately demo map is limited to z9, so we can't see the difference on all the midzoom levels (z10-z11). But I think this still has some visual problems when there are many small patches of green (especially dark green) and some fading might help - Moscow area example:

http://maps.imagico.de/#map=9/55.766/37.711&lang=en&r=osmlz&o=aa&ui=2

![screenshot_2018-11-15 imagico de maps 1](https://user-images.githubusercontent.com/5439713/48555741-5cae9b80-e8e2-11e8-96ff-7a350c731428.png)

It seems however that dark green (for high vegetation) is changed for something lighter, so it needs more attention anyway, but is promising - the same area using (probably) our current forest/wood green without fading (some other colors are not up to date) looks worse:

http://maps.imagico.de/#map=9/55.766/37.711&lang=en&r=osmlz&o=ea&ui=2

![screenshot_2018-11-15 imagico de maps 2](https://user-images.githubusercontent.com/5439713/48555948-d34b9900-e8e2-11e8-8688-fd56ab0e26a2.png)

Another issue is where to draw the line between showing all greens and generalized vegetation greens, because we still want to show specific green areas from some zoom level.

There are also other changes - motorways are turned into violet ( http://blog.imagico.de/more-new-colors/ ), which was possible after border color change into something else - I like both these changes, but it needs even more testing and separate tickets to discuss them.

**Comment by jeisenbe** [comment]
FYI, the low-zoom colors on Imagico's alt-colors fork are:
```
@built-up-lowzoom: #d7d4d3;
@vegetation-tall-lowzoom: #b6dca6;
@vegetation-low-lowzoom: #ddecbd;
@bare_ground-lowzoom: #efe7d9;
```

Compare to:
```
@grass: #cdebb0;
@scrub: #b5e3b5;
@forest: #add19e;  
...
@residential: #e0dfdf; 
...
@bare_ground: #eee5dc;

```
So they are all adjusted compared to the high-zoom colors. 

This is based off of the repository on GitHub, https://github.com/imagico/osm-carto-alternative-colors/tree/master

**Vegetation-low includes:**
picnic_site, caravan_site, camp_site, 
grave_yard, cemetery, 
park, recreation_ground, dog_park, golf_course, miniature_golf,
allotments, farmland, greenhouse_horticulture
meadow/grassland/grass, village_green, common, garden, 
heath, bog, string_bog, wet_meadow, fen, marsh

**Vegetation-tall includes:**
vineyard, orchard, plant_nursery, 
wood, forest, scrub, swamp

**Bare_ground:**
quarry, landfill, 
bare_rock, scree, shingle, sand
beach, shoal, 

**Built-up:**
residential, retail, industrial, commercial,
garages, farmyard, railway, construction, brownfield
power station/generator/substation
hospital, clinic, university, college, school, kindergarten
parking, aeroway=apron, aerodrome, 
highway=services, rest_area
railway=station
fitness_center, sports_center, stadium, track, pitch

I notice that mangroves were left out; I would probably include these under vegetation_tall, and I would also like to give mangroves the new scrub color as a background, once it is decided.

**Comment by jeisenbe** [comment]
Right, I’m suggesting that they could get a fill in the future, similar to
swamp.

But many (most?) mangroves are closer to scrub in height, so the color
could be the same as that used for scrub; this would also help distinguish
swamp and mangroves

**Comment by jeisenbe** [comment]
Related to #3647

As discussed, the three ways to fix this issue are:

1) Remove the landcover fading, and render all landcovers at full strength up to their new limits (z5 for forest, meadow, scrub, heath, farmland, wetland etc). 
- This option is followed by the French branch

2) Change the fading to take place at a higher level, say z7 to z5. This would give the landcover the appearance of "fading out" right before it disappears at z4, which is similar to what is done on some other maps. z8 to z12 would now be in full color. 
- This could also be done somewhat sooner, say z10 to z8, so that z11 and z12 would show still show the full strength colors. 

3) Use a reduced set of colors at low zoom, combining different types of vegetation into high and low vegetation, and showing all developed land the same. We already do the latter at mid-zoom with developed land. This is the method used by @imagico in the alternative-colors branch: eg http://maps.imagico.de/#map=9/55.766/37.711&lang=en&r=osmlz&o=aa&ui=2
- If the forest color is still too dark in this example, we could try something lighter. It's not possible to render the forest color much lighter, or to shift the hue much, when scrub and orchards rendered differently, but with all of these combined in one color there are many options. 

Would some examples of 3) at z10 thru z12 be helpful?

**Comment by kocio-pl** [comment]
I am not sure if combining is what really helps here and if it's needed (it's always better to show the differences by default). I see the difference between forest colors in https://github.com/gravitystorm/openstreetmap-carto/issues/3513#issuecomment-439042799 and this might be more important factor.

Using current OSM Carto color layer (without fading) and Gpicker wood is `#ADD19D` (close to `#ADD19E` - maybe some small Firefox display error?):

http://maps.imagico.de/#map=9/55.722/37.672&lang=en&r=osmlz&o=ea&ui=2

while using alternative colors layer it's much better `#B6DCA5`: 

http://maps.imagico.de/#map=9/55.722/37.672&lang=en&r=osmlz&o=aa&ui=2

which seems to be close to the `vegetation-tall-lowzoom` defined as `#b6dca6` here (probably the same small display error in browser):

https://github.com/imagico/osm-carto-alternative-colors/blob/441d59909d3dd548731789fec9d75200ee606023/landcover.mss#L76-L84

I would check first if changing woods/forest color to `#b6dca6` is not enough on mid/low zoom and how does it work on high zoom.

**Comment by imagico** [comment]
Two comments here:

> This option is followed by the French branch

The french style is not rendering low zoom landcover with mapnik, it uses a pre-rendered image that is scaled to the zoom level in question:

https://github.com/cquest/osmfr-cartocss/blob/master/osmfr.yml#L84-L90
https://github.com/gravitystorm/openstreetmap-carto/pull/2654#issuecomment-314744738

And the one bit mismatch in colors in the low zoom demo is due to some rounding error in color space conversion in the renderer.

**Comment by jeisenbe** [comment]
Points 3, 4, and 5 (worse contrast with scrub, heath, allotments) are
serious problems.

This is not a problem at mid to low zoom level in the alt-colors branch,
where heath and allotments are merged with grass and farmland, and scrub is
merged with forest.

**Comment by Adamant36** [comment]
I think it looks good. Especially with residential. I don't see the issue with scrub either. Although it could be my monitor. Heath and allotments do look worse though, but its not like those can't be tweaked.

**Comment by jeisenbe** [comment]
**Problem 8.**: another issue with`#b6dca6` for woodland/forest at high zoom is that orchard/vineyard color is `#aedfa3;` - these are very similar. 

`#b6dca6` is Lch(84,32,135) and `#aedfa3;` is Lch(84,35,138) - so the lightness is identical, the hue is almost the same, and the chroma is very close.

This is probably one of the reasons why @imagico merged the orchard and forest colors at low zoom.

Here's a test rendering of Southern Australia (near Adelaide), where there are many vineyards and some areas of forest:

https://www.openstreetmap.org/#map=12/-34.6493/138.9740
Master (current rendering) z12
![z12-s-aus-master](https://user-images.githubusercontent.com/42757252/51665877-10798900-2000-11e9-87ae-063806d63d94.png)

z12 Without fading (also same as alt-colors style landcover at this zoom level)
![z12-s-aus-nofade](https://user-images.githubusercontent.com/42757252/51666090-972e6600-2000-11e9-803b-7e899aa9f820.png)

z12 Forest `#b6dca6`
![z12-s-aus-nofade-forest](https://user-images.githubusercontent.com/42757252/51666060-85e55980-2000-11e9-9814-3633f6d424eb.png)
- Very hard to see a difference between vineyard and forest


Master z11
![z11-s-aus-master-12 -34 6493 138 9740](https://user-images.githubusercontent.com/42757252/51666070-8c73d100-2000-11e9-99cd-a2585cf9b84b.png)

z11 Without fading
![z11-s-aus-nofade](https://user-images.githubusercontent.com/42757252/51666239-ca70f500-2000-11e9-9f6f-21460fb5d84d.png)

z11 Forest `#b6dca6`
![z11-s-aus-nofade-forest](https://user-images.githubusercontent.com/42757252/51666196-b0cfad80-2000-11e9-8a5c-939cbe24c024.png)

z11 Alt-colors for landcover
- There are 4 landcover colors: tall vegetation, low vegetation, bare ground, and developed land
![z11-s-aus-altcolors](https://user-images.githubusercontent.com/42757252/51666263-d2c93000-2000-11e9-9383-85d8d5a6631c.png)


Master z10
![z10-s-aus-master](https://user-images.githubusercontent.com/42757252/51666339-f1c7c200-2000-11e9-84ea-59b764167b84.png)

z10 Without fading
![z10-s-aus-nofade](https://user-images.githubusercontent.com/42757252/51666353-f9876680-2000-11e9-8939-2c68b60c9946.png)

z10 Forest `#b6dca6`
![z10-s-aus-nofade-forest](https://user-images.githubusercontent.com/42757252/51666465-3f442f00-2001-11e9-886c-4da74137a8ac.png)

z10 Alt-colors for landcover
![z10-s-aus-altcolors](https://user-images.githubusercontent.com/42757252/51666405-16bc3500-2001-11e9-8edf-52e0cc91bbbc.png)

**Comment by dch0ph** [comment]
@imagico I think the core of this issue was addressed by removing the landuse fading (#5148). The plan in #5148 was then to tackle the individual related issues, such as #3670.

**Comment by imagico** [comment]
Yes, this and #3935 are essentially addressed by #5148 - although there are still plenty of contrast related issues at these scales beyond #3936.  There was a reason - after all - why the fading was initially introduced in #2654.  But this is largely covered by #3647.

So closing this and #3935.
